test/get_previous_releases.py

```diff
@@ -148,6 +148,24 @@ def download_binary(tag, args) -> int:
     ret = subprocess.run(['tar', '-zxf', tarball, '-C', tag,
                           '--strip-components=1',
                           'bitcoin-{tag}'.format(tag=tag[1:])]).returncode
+    
+    if tag >= "v23" and platform == "arm64-apple-darwin":
+        # Starting with v23.0 there are arm64 binaries for ARM (M1/M2) macs, but they have to be signed to run
+        binary_path = '{cwd}/{tag}/bin/bitcoind'.format(cwd=os.getcwd(), tag=tag)
+
+        # Is it already signed?
+        result = subprocess.run(['codesign', '-v', '-d', binary_path], capture_output=True)
+        if "code object is not signed at all" in result.stderr.decode("utf-8"):
```

Comment:
This seems brittle. Is there a return code we can use?

Reply:
Agreed. But success or error (e.g. using an unrecognized flag) both return `1`.

The output ends up in `stderr` in either case, too. Unfortunately.